src/js/components/Button/Button.js

```diff
@@ -281,8 +322,17 @@ const Button = forwardRef(
             theme.global.colors.text[isDarkBackground() ? 'dark' : 'light'],
         });
       }
+    } else if (kind && kindIcon && !plain) {
```

Comment:
Could this just be `} else if (!plain && !icon.props.color) {`?

Reply:
This logic doesn't quite work -- we still needed to have a way to check that a `kindIcon` exists before doing this (otherwise was running into snapshot changes on menu, other button components).

But i did simplify slightly.